src/DGtal/geometry/curves/AlphaThickSegmentComputer.h

```diff
@@ -0,0 +1,711 @@
+   */
+  bool isValid() const;
+  
+ 
+  /**
+   * Get the extremity points of the segment. These points are not
+   * necessary the last point of the segment.
+   *
+   **/
+  std::pair<InputPoint, InputPoint>  getExtremityPoints() const;
+  
+
+  /**
+   * @return the antipodal leaning points of the segment (given in the
+   * convexhull). The result is given as a pair for which the first
+   * element is the pair containing the edge antipodal points and the
+   * second element is the vertex of the antipodal pair (see section \ref
+   * moduleAlphaThickSegmentRecoIntro for an illustration of such an
+   * antipodal pair).
+   **/
+  std::pair<std::pair<InputPoint, InputPoint>, InputPoint>
+  getAntipodalLeaningPoints() const;
+  
+  
+  /**
+   * Computes the paralell strip params from the current state of the segment.
+   * @param[out] mu the minimal value of N.X (with N is the normal vector of the segment).
+   * @param[out] N the normal of the vector (not normalized).
+   * @param[out] nu the width of the strip. 
+   *
+   **/
+  void computeParalellStripParams(double &mu, PointD &N, double &nu) const;
+  
+  
+  /**
+   * @return the two values of the normal vector given as a Point(a,b)
+   * (the normal is oriented toward the segment by considering the
+   * edge PQ of the convexhull antipodal pair)..
+   **/
+  
+  PointD getNormal() const;
+
+  /**
+   * @return the mu parameter of the current segment.
+   **/
+  double getMu() const;
+ 
+ 
+  /**
+   * @return the nu parameter the of the current segment.
+   **/
+  double getNu() const;
+   
+  
+   /**
+   * @return the segment length defined from the bouding box (@see getBoundingBox).
+   **/
+  double getSegmentLength() const;
+  
+
+  /**
+   * @return 'true' if the points of the segment computer are stored in the main container.
+   **/
+  bool isStoringSegmentPoints() const;
+
+
+  /**
+   * @return the total number of points beeing recognized through in the segment construction.
+   **/  
+  unsigned int getNumberSegmentPoints() const;
+
+
+
+  // ------------------------- Display services ------------------------------
+
+
+  /**
+   * @return the current alpha thick segment convexhull given as a vector containing the vertex points.
+   **/
+  std::vector<InputPoint> getConvexHull() const;
+
+
+  
+  /**
+   * Computes the segment bounding box according to two extremity points (\a aFirstPt, \a aLastPt).
+   * @param[in] aFirstPt the first extrem point.
+   * @param[in] aLastPt the last extrem point. 
+   * @param[out] pt1LongestSegment1 the first point of one of the longest segment.
+   * @param[out] pt2LongestSegment1 the second point of one of the longest segment.
+   * @param[out] pt3LongestSegment2 the first point of one of the second longest segment.
+   * @param[out] pt4LongestSegment2 the second point of one of the second longest segment.
+   * @param[in] minVisibleWidthBounds the minimal width of the resulting bounding box (for drawing issue).
```

Comment:
Is this parameter only useful when displaying a segment ? Does it enlarge the bbox ?